test/catch2/live_restore/unit/test_live_restore_extent_list.cpp

```diff
@@ -0,0 +1,152 @@
+ */
+
+#include "../utils_live_restore.h"
+
+using namespace utils;
+
+TEST_CASE("Live Restore Extent Lists: Creation", "[live_restore],[live_restore_extent_list]")
+{
+    /*
+     * Note: this code runs once per SECTION so we're creating a brand new WT database for each
+     * section. If this gets slow we can make it static and manually clear the destination and
+     * source for each section.
+     */
+    live_restore_test_env env;
+
+    WT_SESSION_IMPL *session = env.session;
+    WT_SESSION *wt_session = reinterpret_cast<WT_SESSION *>(session);
+
+    std::string file_name = "MY_FILE.txt";
+    std::string source_file = env.source_file_path(file_name);
+    std::string dest_file = env.dest_file_path(file_name);
+
+    SECTION("Open a new unbacked file")
+    {
+        WT_LIVE_RESTORE_FILE_HANDLE *lr_fh;
+        open_lr_fh(env, dest_file.c_str(), &lr_fh);
+
+        // There's no backing file in the source directory, so no extent list to track.
+        REQUIRE(extent_list_str(lr_fh) == "");
+        lr_fh->iface.close(reinterpret_cast<WT_FILE_HANDLE *>(lr_fh), wt_session);
+    }
+
+    SECTION("Open a new backed file")
+    {
+        create_file(source_file.c_str(), 1000);
+
+        WT_LIVE_RESTORE_FILE_HANDLE *lr_fh;
+        open_lr_fh(env, dest_file.c_str(), &lr_fh);
+
+        // We've created a new file in the destination backed by a file in source.
+        // We haven't read or written anything so the file is one big hole.
+        REQUIRE(extent_list_in_order(lr_fh));
+        REQUIRE(extent_list_str(lr_fh) == "(0-999)");
+        lr_fh->iface.close(reinterpret_cast<WT_FILE_HANDLE *>(lr_fh), wt_session);
+    }
+
+    SECTION("The extent list can't have holes beyond the end of the destination file")
+    {
+        // The following steps aren't a realistic scenario in Live Restore, but it gets
+        // us to the desired test state.
+
+        // Create a source file of size 110 and open the lr_fh to copy that metadata.
+        // This creates a destination file also of size 110.
+        create_file(source_file.c_str(), 110);
+        WT_LIVE_RESTORE_FILE_HANDLE *lr_fh;
+        open_lr_fh(env, dest_file.c_str(), &lr_fh);
+
+        // Replace the source file with a larger file of size 200 and reopen the destination file
+        // to recompute the extent list.
+        testutil_remove(source_file.c_str());
+        create_file(source_file.c_str(), 200);
+
+        lr_fh->iface.close(reinterpret_cast<WT_FILE_HANDLE *>(lr_fh), wt_session);
+        open_lr_fh(env, dest_file.c_str(), &lr_fh);
+
+        // The extent list is capped at the size of the destination file. It doesn't take the
+        // source file size into account.
+        REQUIRE(extent_list_in_order(lr_fh));
+        REQUIRE(extent_list_str(lr_fh) == "(0-109)");
+        lr_fh->iface.close(reinterpret_cast<WT_FILE_HANDLE *>(lr_fh), wt_session);
+    }
+
+    SECTION("The extent list can have holes beyond the end of the source file")
```

Comment:
Can you add one more case here:

Source file size 110
Destination file size 110

Destination is truncated to 50
Extent list must be 0-49

Destination file is positive truncated to 110
Extent list must still be 0-49.

Reply:
Discussed offline. We'll add this when writing the `fh_truncate` API tests.